### ISSUE TITLE
[ Search ] lack of possibility to clear value via keyboard

### ISSUE THREAD

**Issue by DlgSHi** [Sage/carbon#3485]
### Current behaviour

Using a `Search` component we have no possibility to clear value via keyboard.

Steps to reproduce:
1. Open `Design System` -> `Search` -> `default`
2. Type `test` inside the search input
3. Press Tab - to focus `cross` icon

Actual Result:
Cross icon is not focusable and there is no possibility to clear value via keyboard.

### Expected behaviour

Cross icon should be focusable and there should be possibility to clear value via keyboard pressing `Space`/`Enter` over the hovered `cross` icon.

### Reproducible example

[story](https://carbon.sage.com/?path=/story/design-system-search--default-story)
### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 53.0.0 |
| carbon-factory   |
| react-scripts    | 
| React            |
| Browser          | All |
| npm              |
| Operating System | macOS |

**Comment by DlgSHi** [comment]
@harpalsingh what do you think about this ?

**Comment by edleeks87** [comment]
I'll message UX for some guidance on this and create a ticket based on their steer

**Comment by edleeks87** [comment]
Clarified with UX, we should make the `cross`/ `close` icon "tabbable"

FE-3449
